public/app/features/dashboard/components/DashboardSettings/AutoRefreshIntervals.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { FC, useCallback, useEffect, useMemo, useState } from 'react';
+import { Input, Tooltip } from '@grafana/ui';
+import { defaultIntervals } from '@grafana/ui/src/components/RefreshPicker/RefreshPicker';
+
+import { getTimeSrv } from '../../services/TimeSrv';
+
+interface Props {
+  renderCount: number;
+  refreshIntervals: string[];
+  onRefreshIntervalChange: (interval: string[]) => void;
+}
+
+export const AutoRefreshIntervals: FC<Props> = ({ renderCount, refreshIntervals, onRefreshIntervalChange }) => {
+  const [intervals, setIntervals] = useState<string[]>(
+    getTimeSrv().getValidIntervals(refreshIntervals ?? defaultIntervals)
+  );
+  const [invalidIntervalsMessage, setInvalidIntervalsMessage] = useState<string | null>(null);
+
+  useEffect(() => {
+    const intervals = getTimeSrv().getValidIntervals(refreshIntervals ?? defaultIntervals);
+    setIntervals(intervals);
+  }, [renderCount, refreshIntervals]);
+
+  const intervalsString = useMemo(() => {
+    if (!Array.isArray(intervals)) {
+      return '';
+    }
+
+    return intervals.join(',');
+  }, [intervals]);
+
+  const onIntervalsChange = useCallback(
+    (event: React.FormEvent<HTMLInputElement>) => {
+      if (!event.currentTarget.value) {
+        return;
+      }
+
+      const newIntervals = event.currentTarget.value.split(',');
+      const invalidMessage = validateIntervals(newIntervals);
+
+      setIntervals(newIntervals);
+      setInvalidIntervalsMessage(invalidMessage);
+    },
+    [setIntervals, setInvalidIntervalsMessage]
+  );
+
+  const onIntervalsBlur = useCallback(
+    (event: React.FormEvent<HTMLInputElement>) => {
+      if (!event.currentTarget.value || invalidIntervalsMessage) {
+        return;
+      }
+
+      onRefreshIntervalChange(getValidIntervals(intervals));
+    },
+    [intervals, onRefreshIntervalChange]
+  );
```

Comment:
Can't see much benefit of live(on change) validation - maybe onBlur would be sufficient?

Reply:
Thank you, I agree and will change it back.